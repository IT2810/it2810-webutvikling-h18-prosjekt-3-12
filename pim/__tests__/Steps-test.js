import "react-native";
import React from "react";
import Steps from "../screens/StepsScreen";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";

describe("Steps snapshot", () => {
  it("should render steps view correctly", () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<Steps />);
    expect(result).toMatchSnapshot();
  });

  it("should render outer most view in steps view correctly", () => {
    //tests a snapshot view
    const tree = renderer.create(<Steps />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

//tests component functions
let stepsContainer = renderer.create(<Steps />).getInstance();
let formatedDate = stepsContainer.formatDate();
console.log(formatedDate);
