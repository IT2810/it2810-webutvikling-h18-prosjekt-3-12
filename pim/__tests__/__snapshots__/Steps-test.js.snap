// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps snapshot should render outer most view in steps view correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
/>
`;

exports[`Steps snapshot should render steps view correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "paddingTop": 30,
      }
    }
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "marginHorizontal": 40,
        }
      }
    >
      <Component
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "paddingBottom": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          Activity
        </Text>
        <Icon
          allowFontScaling={false}
          name="ios-walk"
          size={30}
          style={
            Object {
              "marginLeft": 10,
              "marginTop": -4,
            }
          }
        />
      </Component>
      <Component
        style={
          Object {
            "backgroundColor": "#E9F7FD",
            "borderColor": "#d6d7da",
            "borderRadius": 7,
            "borderWidth": 0.5,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 15,
            "overflow": "hidden",
            "padding": 20,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(96,100,109, 1)",
              "fontSize": 17,
              "lineHeight": 24,
            }
          }
        >
          Current steps: 
          0
           steps
        </Text>
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "marginTop": 5,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="ios-information-circle-outline"
            size={16}
            style={
              Object {
                "color": "rgba(96,100,109, 1)",
              }
            }
          />
        </TouchableOpacity>
      </Component>
      <Component
        style={
          Object {
            "backgroundColor": "#E9F7FD",
            "borderColor": "#d6d7da",
            "borderRadius": 7,
            "borderWidth": 0.5,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 15,
            "overflow": "hidden",
            "padding": 20,
            "paddingBottom": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(96,100,109, 1)",
              "fontSize": 17,
              "lineHeight": 24,
            }
          }
        >
          Past day: 
          0
           steps
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(96,100,109, 1)",
              "fontSize": 11,
              "marginVertical": 15,
            }
          }
        >
          Oct 17, 14:50
        </Text>
      </Component>
      <Component
        style={
          Object {
            "backgroundColor": "#E9F7FD",
            "borderColor": "#d6d7da",
            "borderRadius": 7,
            "borderWidth": 0.5,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 15,
            "overflow": "hidden",
            "padding": 20,
            "paddingBottom": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(96,100,109, 1)",
              "fontSize": 17,
              "lineHeight": 24,
            }
          }
        >
          Distance: 
          0 m
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(96,100,109, 1)",
              "fontSize": 11,
              "marginVertical": 15,
            }
          }
        >
          Oct 17, 14:50
        </Text>
      </Component>
      <Component
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginTop": 10,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "#ff837c",
                "borderRadius": 7,
                "color": "white",
                "fontSize": 12,
                "lineHeight": 24,
                "overflow": "hidden",
                "padding": 5,
              }
            }
          >
            Reset data
          </Text>
        </TouchableOpacity>
      </Component>
    </Component>
  </ScrollViewMock>
</Component>
`;
